bin/pkg.ml

```diff
@@ -227,33 +227,90 @@ module Lock = struct
         ~all_contexts_arg:all_contexts
         ~version_preference_arg:version_preference
     in
-    let* source_dir = Memo.run (Source_tree.root ()) in
-    let project = Source_tree.Dir.project source_dir in
-    let dune_package_map = Dune_project.packages project in
-    let opam_file_map = opam_file_map_of_dune_package_map dune_package_map in
-    let+ sys_env =
-      Dune_pkg.Sys_poll.sys_env ~path:(Env_path.path Stdune.Env.initial)
+    let* sys_bindings_from_current_system =
+      if use_env_from_current_system then
+        Dune_pkg.Sys_poll.sys_bindings ~path:(Env_path.path Stdune.Env.initial)
+      else Fiber.return Dune_pkg.Solver_env.Sys_var.Bindings.empty
     in
-    let repo_selection =
-      Dune_pkg.Opam.Repo_selection.add_env sys_env repo_selection
-    in
-    (* Construct a list of thunks that will perform all the file IO side
-       effects after performing validation so that if materializing any
-       lockdir would fail then no side effect takes place. *)
-    let write_disk_list =
+
+    if just_print_solver_env then
+      Fiber.return
+      @@ List.iter per_context
+           ~f:(fun
+                { Per_context.solver_env = solver_env_from_context
+                ; context_common = { name = context_name; _ }
+                ; _
+                }
+              ->
+             let solver_env =
+               merge_current_system_bindings_into_solver_env_from_context
+                 ~context_name ~solver_env_from_context
+                 ~sys_bindings_from_current_system
+             in
+             User_message.make
+               [ Pp.textf "Solver environment for context %s:\n%s"
+                   (String.maybe_quoted
+                   @@ Dune_engine.Context_name.to_string context_name)
+                   (Dune_sexp.to_string @@ Dune_pkg.Solver_env.encode solver_env)
+               ]
+             |> Console.print_user_message)
```

Comment:
`Console.print` will eliminate a little indentation

Reply:
Ended up getting rid of the list and using `Console.printf`